manifests-gen/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+
+	certmangerv1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+)
+
+var basePath = flag.String("base-path", "", "path to the root of the provider's repository")
+var manifestsPath = flag.String("manifests-path", "", "path to the desired directory where to output the generated manifests")
+var providerName = flag.String("provider-name", "", "name of the provider")
+var providerType = flag.String("provider-type", "", "type of the provider")
+var providerVersion = flag.String("provider-version", "", "version of the provider")
+var projDir string
```

Comment:
Why not declare as a `var ( )` block? Do they even need to be declared as globally scoped?